src/Main.elm

```diff
@@ -1,10 +1,829 @@
+                ]
+                [ upperPanel
+                , E.column
+                    [ E.width E.fill
+                    , E.height E.fill
+                    , E.spacing 1
+                    ]
+                    -- folders first, just like "FIRST_STEP.md" says
+                    (filteredAudienceFolders ++ filteredAudiences)
+                ]
+    in
+    E.row
+        [ E.spacing 20
+        ]
+        [ E.el [ E.alignTop, E.width <| E.px 700 ] browser
+        , viewSortingPanel model.sortBy
+        ]
+
+
+viewFolder : Int -> FormattedAudienceFolder -> List (E.Element Msg) -> List (E.Element Msg)
+viewFolder id folder acc =
+    let
+        -- we view a "folder" as a row: the icon, and the name of the folder
+        item =
+            E.row
+                [ E.paddingXY 15 10
+                , EBackground.color Palette.gray5
+                , EEvents.onClick (FolderClicked id)
+                , EFont.color Palette.gray0
+                , E.width E.fill
+                , E.spacing 15
+                , E.htmlAttribute <| Html.Attributes.style "cursor" "pointer"
+                , E.mouseOver <| [ EBackground.color <| E.rgb255 255 255 255 ]
+                ]
+                [ E.el
+                    [ E.width <| E.px 18
+                    , E.height <| E.px 18
+                    ]
+                    <| E.html <| Icons.folder "#6ea4fc"
+                , E.text folder.name
+                ]
+    in
+    item :: acc
+
+
+viewAudience : Int -> FormattedAudience -> List (E.Element Msg) -> List (E.Element Msg)
+viewAudience _ audience acc =
+    let
+        -- we view an "audience" as a row: the icon, and the name of the audience
+        item =
+            E.row
+                [ E.paddingXY 15 10
+                , EBackground.color Palette.gray5
+                , E.width E.fill
+                , E.spacing 15
+                , EFont.color Palette.gray0
+                , E.mouseOver <| [ EBackground.color <| E.rgb255 255 255 255 ]
+                ]
+                [ E.el
+                    [ E.width <| E.px 17
+                    , E.height <| E.px 17
+                    ]
+                    <| E.html <| Data.Audience.audienceTypeToIcon audience.type_
+                , E.text audience.name
+                ]
+    in
+    item :: acc
+
+
+viewColor : E.Color -> E.Element Msg
+viewColor c =
+    E.el
+        [ E.width E.fill
+        , E.height <| E.px 100
+        , EBackground.color c
+        , E.alignBottom
+        ]
+        E.none
+
+
+
+-- this just wraps some element in a colored circle
+wrapColorCircle : E.Color -> E.Element Msg -> E.Element Msg
+wrapColorCircle color_ content_ =
+    E.el
+        [ E.padding 5
+        , EBorder.rounded 99999
+        , EBackground.color color_
+        ]
+        content_
+
+
+colors : List E.Color
+colors =
+    [ Palette.color0
+    , Palette.color1
+    , Palette.color2
+    , Palette.color3
+    , Palette.color4
+    , Palette.color5
+    ]
```

Comment:
I'd expect this to be in Palette module as well...

Reply:
The reason I didn't is because this is only used in the Main.elm file. If we had other files, and they wanted to have a list of colors, I think those lists should be put in the module that uses them.

My thought process was "what if we had a module that instead of needing a list of all colors from the Palette module, it only needed `[Palette.color1, Palette.color3, Palette.color5]`? Or it needed them in a different order? Should that belong in Palette as well?"

This is why I concluded that a list of colors belongs in the module that actually uses it.